tests/test_hidpp20.py

```diff
@@ -39,3 +39,43 @@ def device(self, hidpp, fd):
         yield dev
 
         hidpp.hidpp20_device_destroy(dev)
+
+    # IRoot
+    def test_root_get_feature(self, client, id, hidpp, device):
+        feature_index = hidpp.new_uint8_tp()
+        feature_type = hidpp.new_uint8_tp()
+        feature_version = hidpp.new_uint8_tp()
+
+        # IFeatureSet
+        assert not hidpp.hidpp_root_get_feature(device, 0x0001, feature_index, feature_type, feature_version)
+
+        assert hidpp.uint8_tp_value(feature_index) == 1
+        assert hidpp.uint8_tp_value(feature_type) == 0
+        assert hidpp.uint8_tp_value(feature_version) == 1
+
+        # DeviceInformation
+        assert not hidpp.hidpp_root_get_feature(device, 0x0003, feature_index, feature_type, feature_version)
+
+        assert hidpp.uint8_tp_value(feature_index) == 2
+        assert hidpp.uint8_tp_value(feature_type) == 0
+        assert hidpp.uint8_tp_value(feature_version) == 2
+
+        # BatteryVoltage
+        assert not hidpp.hidpp_root_get_feature(device, 0x1001, feature_index, feature_type, feature_version)
```

Comment:
I agree IFeatureSet and DeviceInformation have a good chance of being placed at indexes 1 and 2, but BatteryVoltage should probably not be hardcoded in a generic test for the root feature.

Reply:
But we control the test device. In this case we are creating a G Pro and checking to see if the `0x1001` feature is in the correct slot.

Reply:
yes, but no. The test device here is the G Pro, but if you want to run the same test (because you subclass this class into a G900, you will have situations where the battery is not at index 6.

So generic tests should not have specifics, and specifics should be in subclasses (here the `TestGPro` class, which would call the common `TestHIDPP` class code, and then add specialization.

Reply:
The class it's not really prepared to be subclassed, I was going to deal with that when we needed, but it looks like it is going to be in this PR.

Reply:
Same. This is out of scope for this PR, I added a `TODO` in the code.